### PR TITLE
Integration tests for example project structure

### PR DESCRIPTION
**Pull request by o-tsaruk**
Created integration tests for `example_project_structure` contracts;
Fixed `repay` method;
Added `get_asset_shares` method.

### CODE REVIEW

File: example_project_structure/impls/lending/lending.rs [hunk 1]
```diff
@@ -60,6 +60,13 @@ impl<T: LendingStorage<Data = LendingData> + PausableStorage<Data = PausableData
         Ok(PSP22Ref::total_supply(&mapped_asset))
     }
 
+    default fn get_shares_from_asset(&self, asset_address: AccountId) -> Result<AccountId, LendingError> {
```
Comment (coreggon11) [inline example_project_structure/impls/lending/lending.rs:63]:
```suggestion
    default fn get_asset_shares(&self, asset_address: AccountId) -> Result<AccountId, LendingError> {
```

Not meant to be mean :smiley: but you want to get the share token of this asset not asset from the shares

File: tests/e2e/example_project_structure/lending_contract.tests.ts [hunk 2]
```diff
@@ -1,37 +1,223 @@
...
 
 describe('LENDING_CONTRACT', () => {
   async function setup() {
-    const stable_coin = await setupContract('stable_coin_contract', 'new', 'Green Coin', 'GC')
-    const loan = await (await setupContract('loan_contract', 'new' )).abi
+    const greencoin = await setupContract('stable_coin_contract', 'new', 'Green Coin', 'GC')
+    const loan = await (await setupContract('loan_contract', 'new')).abi
     const shares = await (await setupContract('shares_contract', 'new', '', '')).abi
     const lending = await setupContract('lending_contract', 'new', shares.source.hash, loan.source.hash)
+    const redcoin = await setupContract('stable_coin_contract', 'new', 'Red Coin', 'RC')
```
Comment (coreggon11) [inline tests/e2e/example_project_structure/lending_contract.tests.ts:15]:
```suggestion
    const sashacoin = await setupContract('stable_coin_contract', 'new', 'Red Coin', 'RC')
```

Comment (coreggon11) [reply tests/e2e/example_project_structure/lending_contract.tests.ts:15]: or nameless (its the name of the token) :joy:

Comment (coreggon11) [reply tests/e2e/example_project_structure/lending_contract.tests.ts:15]: we will also need krikocoin

File: tests/e2e/example_project_structure/lending_contract.tests.ts [hunk 3]
```diff
@@ -1,37 +1,223 @@
...
 
-    return { lending, stable_coin, alice: stable_coin.defaultSigner }
+  function result(s: string | undefined) {
+    const result: Result = s != null ? JSON.parse(s) : null
+    return result
   }
 
+  async function borrow(lendingContract, borrowedToken, collateralToken, user, approveAmount, borrowedAmount, price) {
+    // collateralToken approves amount of tokens for lending contact
+    await expect(collateralToken.tx.approve(lendingContract.address, approveAmount)).to.eventually.be.fulfilled
```
Comment (coreggon11) [inline tests/e2e/example_project_structure/lending_contract.tests.ts:27]: to.eventually.be.fulfilled is useless, you can remove it from this test and from other tests in another PR

**Review by coreggon11 — APPROVED** [review]
GJ

File: docs/docs/smart-contracts/example/shares.md [hunk 4]
```diff
@@ -104,28 +104,17 @@ We will be using these extensions in our token, so we will implement them for
 our storage.
 
 ```rust
-/// Implement PSP22 Trait for our coin
-impl PSP22 for StableCoinContract {}
+    // Implement PSP22 Trait for our share
```
Comment (xgreenx) [inline docs/docs/smart-contracts/example/shares.md:107]:
```suggestion
/// Implement PSP22 Trait for our share
```

File: docs/docs/smart-contracts/example/shares.md [hunk 5]
```diff
@@ -104,28 +104,17 @@ We will be using these extensions in our token, so we will implement them for
...
-            instance.metadata.name = name;
-            instance.metadata.symbol = symbol;
-            instance.metadata.decimals = 18;
-            let total_supply = 1_000_000 * 10_u128.pow(18);
-            assert!(instance._mint(instance.env().caller(), total_supply).is_ok());
-        })
-    }
-}
+// It forces the compiler to check that you implemented all super traits
+impl Shares for SharesContract {}
```
Comment (xgreenx) [inline docs/docs/smart-contracts/example/shares.md:117]: Why did you remove constuctor?=)

File: docs/docs/smart-contracts/example/impls.md [hunk 6]
```diff
@@ -264,35 +268,27 @@ where
...
-        .data()
-        .shares_asset
-        .get(shares_address)
-        .unwrap_or(ZERO_ADDRESS.into());
-    if token.is_zero() {
-        return Err(LendingError::AssetNotSupported)
-    }
-    Ok(token)
+  where
+          T: Storage<Data>,
```
Comment (xgreenx) [inline docs/docs/smart-contracts/example/impls.md:285]:
```suggestion
where
    T: Storage<Data>,
```

File: docs/docs/smart-contracts/example/impls.md [hunk 7]
```diff
@@ -264,35 +268,27 @@ where
 /// Internal function which will return the address of the shares token
 /// which are minted when `asset_address` is borrowed
 pub fn get_reserve_asset<T>(instance: &T, asset_address: &AccountId) -> Result<AccountId, LendingError>
-where
-    T: Storage<Data>,
+  where
+          T: Storage<Data>,
```
Comment (xgreenx) [inline docs/docs/smart-contracts/example/impls.md:272]:
```suggestion
where
    T: Storage<Data>,
```

**Review by xgreenx — COMMENTED** [review]
And seems you didn't update all docs=)

File: docs/docs/smart-contracts/example/shares.md (hunk 5 above)
Comment (o-tsaruk) [reply docs/docs/smart-contracts/example/shares.md:117]: we create constructor at the last step)

**Review by xgreenx — APPROVED** [review]
